fastai/train.py

```diff
@@ -94,8 +94,52 @@ def clip_grad(learn:Learner, clip:float=0.1)->Learner:
     "Add gradient clipping of `clip` during training."
     learn.callback_fns.append(partial(GradientClipping, clip=clip))
     return learn
-
 Learner.clip_grad = clip_grad
+     
+class AccumulateStepper(LearnerCallback):
+    """
+    Does accumlated step every nth step by accumulating gradients
+    """
+    def __init__(self, learn:Learner, n_step:int = 1):
+        super().__init__(learn)
+        self.n_step = n_step
+ 
+    def on_train_begin(self, **kwargs):
+        "check if loss is reduction"
+        if hasattr(self.loss_func, "reduction") and (self.loss_func.reduction == "mean"):
+             print("For better gradients consider 'reduction=sum'")
+        
+    def on_epoch_begin(self, **kwargs):
+        "init samples and batches, change optimizer"
+        self.acc_samples = 0
+        self.acc_batches = 0
+        
+    def on_batch_begin(self, last_input, last_target, **kwargs):
+        "accumulate samples and batches"
+        self.acc_samples += last_input.shape[0]
+        self.acc_batches += 1
+        
+    def on_backward_end(self, **kwargs):
+        "step if number of desired batches accumulated, reset samples"
+        if (self.acc_batches % self.n_step) == 0:
+            for p in (self.learn.model.parameters()):
+                if p.requires_grad: p.grad.div_(self.acc_samples)
+            self.acc_samples = 0
+            return False
+        else: return True
+    
+    def on_step_end(self, **kwargs):
+        "zero gradients after stepping"
+        if (self.acc_batches % self.n_step) == 0: return False
```

Comment:
Just returning the result of the test is enough ;)

Reply:
I didn't understand this part, can you elaborate please :)

Reply:
No I was saying, `return (self.acc_batches % self.n_step) != 0` (we need `False` when we should do the step + zero, `True` if we skip)

Instead of testing if bool return False else return True, just return not book ;)